### PR TITLE
bug(refs DPLAN-2071): Escape special characters for searching

### PR DESCRIPTION
**Pull request by gruenbergerdemos**
Ticket: https://demoseurope.youtrack.cloud/issue/DPLAN-2071/Text-fehlt-wenn-die-Verfahrenssuche-ein-enthalt

Escape whitspaces and `+` for searching the table.

### CODE REVIEW

File: src/components/DpDataTable/DpDataTable.vue [hunk 1]
```diff
@@ -497,6 +497,23 @@ export default {
...
+     * This method is used to escape special characters in a string.
+     * It specifically targets spaces and plus signs.
+     *
+     * @param {string} string - The string to be processed.
+     * @returns {string} - The processed string with special characters escaped.
+     */
+    escapeSpecialCharacters (string) {
+      return string.replace(/[\s+]/g, (match) => {
+        if (match === ' ') {
+          return '\\s*';
```
Comment (salisdemos) [inline src/components/DpDataTable/DpDataTable.vue:510]: I actually don't now what "You" (the person before You) wants to archive with that. Can You give me an example input with the expected output?

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/DpDataTable/DpDataTable.vue (hunk 1 above)
Comment (gruenbergerdemos) [reply src/components/DpDataTable/DpDataTable.vue:510]: I thought it was used to escape whitespaces eg. `Test test`.

Comment (salisdemos) [reply src/components/DpDataTable/DpDataTable.vue:510]: That doesn't make sense to me. I could imagine that we want to replace special whitespaces to regular whitespaces. That would be something like `.replace(/\s/g, ' ')` . A bit the other way round ;-)

Comment (salisdemos) [reply src/components/DpDataTable/DpDataTable.vue:510]: if so  i suggest something like
```
 escapeSpecialCharacters (string) {
      return string
        .replace(/(\s)/g, ' ')
        .replace(/([+])/g, '\\$&') // here we can add other special chars if necessary without too much adjustment
```

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/DpDataTable/DpDataTable.vue (hunk 1 above)
Comment (gruenbergerdemos) [reply src/components/DpDataTable/DpDataTable.vue]: Done in [064afd7](https://github.com/demos-europe/demosplan-ui/pull/891/commits/064afd77bc01700fbf76a78d16846774eb4c1b73)

**Review by salisdemos — APPROVED** [review]

**Pull request merged**